src/babylon/tools/data/CuckooHash.tsx

```diff
@@ -0,0 +1,369 @@
+        gl.texImage2D(gl.TEXTURE_2D, 0, gl.R16UI, width, height, 0, gl.RED_INTEGER, gl.UNSIGNED_SHORT, data);
+      },
+      texSubImage2D: function(gl, x, y, val) {
+        gl.texSubImage2D(gl.TEXTURE_2D, 0, x, y, 1, 1, gl.RED_INTEGER, gl.UNSIGNED_SHORT, new Uint16Array([val]));
+      },
+    },
+    {
+      dataType: DataType.UInt32,
+      bytePerElement: 4,
+
+      getBufferValue: function(dataview, pos) {
+        return [dataview.getUint32(4 * pos, true), 0];
+      },
+      setBufferValue: function(dataview, pos, val) {
+        dataview.setUint32(4 * pos, val, true);
+      },
+      texImage2D: function(gl, width, height, data) {
+        gl.texImage2D(gl.TEXTURE_2D, 0, gl.R32UI, width, height, 0, gl.RED_INTEGER, gl.UNSIGNED_INT, data);
+      },
+      texSubImage2D: function(gl, x, y, val) {
+        gl.texSubImage2D(gl.TEXTURE_2D, 0, x, y, 1, 1, gl.RED_INTEGER, gl.UNSIGNED_INT, new Uint32Array([val]));
+      },
+    },
+    {
+      dataType: DataType.UInt64,
+      bytePerElement: 8,
+
+      getBufferValue: function(dataview, pos) {
+        return [dataview.getUint32(8 * pos, true), dataview.getUint32(8 * pos + 4, true)];
+      },
+      setBufferValue: function(dataview, pos, lo, hi) {
+        dataview.setUint32(8 * pos, lo, true);
+        dataview.setUint32(8 * pos + 4, hi || 0, true);
+      },
+      texImage2D: function(gl, width, height, data) {
+        gl.texImage2D(gl.TEXTURE_2D, 0, gl.RG32UI, width, height, 0, gl.RG_INTEGER, gl.UNSIGNED_INT, data);
+      },
+      texSubImage2D: function(gl, x, y, lo, hi) {
+        gl.texSubImage2D(gl.TEXTURE_2D, 0, x, y, 1, 1, gl.RG_INTEGER, gl.UNSIGNED_INT, new Uint32Array([lo, hi || 0]));
+      },
+    },
+    {
+      dataType: DataType.Float32,
+      bytePerElement: 4,
+
+      getBufferValue: function(dataview, pos) {
+        return [dataview.getFloat32(4 * pos, true), 0];
+      },
+      setBufferValue: function(dataview, pos, val) {
+        dataview.setFloat32(4 * pos, val, true);
+      },
+      texImage2D: function(gl, width, height, data) {
+        gl.texImage2D(gl.TEXTURE_2D, 0, gl.R32F, width, height, 0, gl.RED, gl.FLOAT, data);
+      },
+      texSubImage2D: function(gl, x, y, val) {
+        gl.texSubImage2D(gl.TEXTURE_2D, 0, x, y, 1, 1, gl.RED, gl.FLOAT, new Float32Array([val]));
+      },
+    },
+  ];
+
+  private _texture: BABYLON.Texture;
+  private _type: DataTypeProperties;
+  private _sizes: HashTableSize;
+  private _setElements: number;
+  private _lastInsertError: { lo: number, hi: number };
+  private _table: ArrayBuffer;
+  private _tableView: DataView;
+  private _hashfunctions: [Hash, Hash];
+
+  constructor(private _scene: BABYLON.Scene, valueType: DataType, private _hashFuncA: hashtype = 'identity',
+              private _hashFuncB: hashtype = 'fnv1a') {
+    Cuckoo._instance = (Cuckoo._instance === undefined) ? 0 : ++Cuckoo._instance;
+
+    this._texture = new BABYLON.Texture('', this._scene, true);
```

Comment:
let's switch away from underscores, try to be more consistent with https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines
(though we already agreed to use single quotes)